cmd/loadTest.go

```diff
@@ -186,22 +194,47 @@ func loadTest(name string, setup setupFunc, test testFunc, cleanup cleanupFunc)
 	close(result)
 	wg2.Wait()
 	ticker.Stop()
+	printQpsWg.Wait()
 
 	sendDuration := lastTick.Sub(t0)
 	testDuration := t1.Sub(t0)
 
-	fmt.Printf("        Warmup: %v queries, %v\n", len(warmups), summarizeTimings(warmups))
-	fmt.Printf("          Test: %v queries, %v\n", len(tests), summarizeTimings(tests))
-	fmt.Printf(" Test duration: %v (%0.2f QPS)\n", testDuration, float64(len(tests))/testDuration.Seconds())
-	fmt.Printf(" Send duration: %v (%0.2f QPS)\n", sendDuration, float64(len(tests))/sendDuration.Seconds())
-	fmt.Printf("Profiling data: %v examples\n", len(profilingDataStrArr))
+	testResult := TestResult{
+		Warmup:             StatisticFromDurations(warmups, warmupDuration),
+		Test:               StatisticFromDurations(tests, testDuration),
+		ActualTestDuration: testDuration,
+		SendDuration:       sendDuration,
+		ProfilingResults:   nil,
+	}
 
-	if len(profilingDataStrArr) != 0 {
-		dataArr := parseProfilingDataStrArr(profilingDataStrArr)
-		summarizeProfilingData(dataArr)
+	if len(profilingMetricStrArr) != 0 {
+		dataArr := parseProfilingMetricStrArr(profilingMetricStrArr)
+		testResult.ProfilingResults = getProfilingMetrics(dataArr)
 		if storeProfilingData {
-			saveProfilingDataToCSV(dataArr)
+			saveProfilingMetricsToCSV(dataArr)
+		}
+	}
+
+	testSummary := &TestSummary{
+		TestTime: testTime.Format(time.RFC3339),
+		Config:   &testConfig,
+		Result:   &testResult,
+	}
+
+	switch outputFormat {
+	case Plain:
+		err := testSummary.WritePlain(os.Stdout)
+		if err != nil {
+			log.Fatalf("failed to write test summary in plain: %v\n", err)
+		}
+	case JSON:
+		err := testSummary.WriteJson(os.Stdout)
+		if err != nil {
+			log.Fatalf("failed to write test summary in json: %v\n", err)
 		}
+	case YAML:
+		log.Fatalf("write test summary in yaml is not yet supported\n")
```

Comment:
Should we raise another tracking issue for this?

Reply:
Sure!

Reply:
Created https://github.com/fortanix/dsm-perf-tool/issues/19

Reply:
Awesome!